dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -1051,9 +1050,22 @@ else if ( isDouble && !Double.isNaN( (Double) value ) )
             }
             else if ( value instanceof BigDecimal )
             {
-                // toPlainString method prevents scientific notation (3E+2)
-                grid.addValue(
-                    ((BigDecimal) value).setScale( 2, RoundingMode.CEILING ).stripTrailingZeros().toPlainString() );
+                Optional<QueryItem> queryItem = params.getItems()
+                    .stream()
+                    .filter( item -> item.isProgramIndicator() && header.getName().equals( item.getItemName() ) )
+                    .findFirst();
+
+                if ( queryItem.isPresent() )
+                {
+                    grid.addValue( AnalyticsUtils.getRoundedValue( params,
+                        ((ProgramIndicator) queryItem.get().getItem()).getDecimals(),
+                        ((BigDecimal) value).doubleValue() ) );
```

Comment:
should this not still have .stripTrailingZeros().toPlainString() ? or is that returned from getRoundedValue?